File: app/src/main/java/com/example/calculadora/MainActivity.java
package com.example.calculadora;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private EditText num1EditText, num2EditText;
    private TextView resultTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        num1EditText = findViewById(R.id.et_primernum);
        num2EditText = findViewById(R.id.et_segnum);
        resultTextView = findViewById(R.id.et_resultado);

        Button addBtn = findViewById(R.id.suma);
        addBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                performOperation('+');
            }
        });

        Button subtractBtn = findViewById(R.id.resta);
        subtractBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                performOperation('-');
            }
        });

        Button multiplyBtn = findViewById(R.id.multiplicacion);
        multiplyBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                performOperation('*');
            }
        });

        Button divideBtn = findViewById(R.id.division);
        divideBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                performOperation('/');
            }
        });
    }

    private void performOperation(char operator) {
        double num1 = Double.parseDouble(num1EditText.getText().toString());
        double num2 = Double.parseDouble(num2EditText.getText().toString());
        double result = 0;

        switch (operator) {
            case '+':
                result = num1 + num2;
                break;
            case '-':
                result = num1 - num2;
                break;
            case '*':
                result = num1 * num2;
                break;
            case '/':
                if (num2 != 0) {
                    result = num1 / num2;
                } else {
                    resultTextView.setText("Error: División entre cero");
                    return;
                }
                break;
        }

        resultTextView.setText(String.valueOf(result));
    }
}

File: app/src/main/java/com/example/calculadora/MainActivity.kt
    package com.example.calculadora
    
    import android.os.Bundle
    import android.widget.Button
    import android.widget.EditText
    import android.widget.TextView
    import androidx.appcompat.app.AlertDialog
    import androidx.appcompat.app.AppCompatActivity
    
    class MainActivity : AppCompatActivity() {
        private var num1EditText: EditText? = null
        private var num2EditText: EditText? = null
        private var resultTextView: TextView? = null
        override fun onCreate(savedInstanceState: Bundle?) {
            super.onCreate(savedInstanceState)
            setContentView(R.layout.activity_main)
            num1EditText = findViewById(R.id.et_primernum)
            num2EditText = findViewById(R.id.et_segnum)
            resultTextView = findViewById(R.id.et_resultado)
            val addBtn = findViewById<Button>(R.id.suma)
            addBtn.setOnClickListener {
                if (validateFields()) {
                    performOperation('+')
                } else {
                    showAlert("Por favor, complete ambos campos.")
                }
            }
            val subtractBtn = findViewById<Button>(R.id.resta)
            subtractBtn.setOnClickListener {
                if (validateFields()) {
                    performOperation('-')
                } else {
                    showAlert("Por favor, complete ambos campos.")
                }
            }
            val multiplyBtn = findViewById<Button>(R.id.multiplicacion)
            multiplyBtn.setOnClickListener {
                if (validateFields()) {
                    performOperation('*')
                } else {
                    showAlert("Por favor, complete ambos campos.")
                }
            }
            val divideBtn = findViewById<Button>(R.id.division)
            divideBtn.setOnClickListener {
                if (validateFields()) {
                    performOperation('/')
                } else {
                    showAlert("Por favor, complete ambos campos.")
                }
            }
            val exitBtn = findViewById<Button>(R.id.exitBtn)
            exitBtn.setOnClickListener { finish() }
        }
    
        private fun validateFields(): Boolean {
            val num1 = num1EditText!!.text.toString().trim { it <= ' ' }
            val num2 = num2EditText!!.text.toString().trim { it <= ' ' }
            return !num1.isEmpty() && !num2.isEmpty()
        }
    
        private fun performOperation(operator: Char) {
            val num1 = num1EditText!!.text.toString().toDouble()
            val num2 = num2EditText!!.text.toString().toDouble()
            var result = 0.0
            when (operator) {
                '+' -> result = num1 + num2
                '-' -> result = num1 - num2
                '*' -> result = num1 * num2
                '/' -> if (num2 != 0.0) {
                    result = num1 / num2
                } else {
                    resultTextView!!.text = "Error: División entre cero"
                    return
                }
            }
            resultTextView!!.text = result.toString()
        }
    
        private fun showAlert(message: String) {
            val builder = AlertDialog.Builder(this)
            builder.setTitle("Alerta")
                .setMessage(message)
                .setPositiveButton("Aceptar", null)
                .show()
        }
    }